tuf-spec.md

```diff
@@ -1185,14 +1201,13 @@ non-volatile storage as FILENAME.EXT.
     metadata file, discard it, abort the update cycle, and report the potential
     rollback attack.
 
-    * **3.3.3**. The version number of the targets metadata file, and all
-    delegated targets metadata files (if any), in the trusted snapshot metadata
-    file, if any, MUST be less than or equal to its version number in the new
-    snapshot metadata file. Furthermore, any targets metadata filename that was
-    listed in the trusted snapshot metadata file, if any, MUST continue to be
-    listed in the new snapshot metadata file.  If any of these conditions are
-    not met, discard the new snaphot metadadata file, abort the update cycle,
-    and report the failure.
+    * **3.3.3**. The version number of the top-level targets metadata file, in
+    the trusted snapshot metadata file, if any, MUST be less than or equal to
+    its version number in the new snapshot metadata file. Furthermore, any
+    targets metadata filename that was listed in the trusted snapshot metadata
+    file, if any, MUST continue to be listed in the new snapshot metadata file.
```

Comment:
Hold on, why are we not checking that every version number for every targets metadata file (including delegations) in the previous snapshot, if any, is <= the current metadata file?

Reply:
It's described in detail in the corresponding commit message (f50f522b86e92f3f827e7af647ec67df4ac41ce1). 

The TLDR is, at this point in the process we don't know if any delegated targets keys were removed and we thus need to skip the rollback check.

Reply:
Hmmm, this basically breaks the workflow described in the Mercury paper, unless we're doing it elsewhere.

Reply:
The check in 4.5.2.7 looks for rollback attacks on delegated targets. Does that cover the workflow from the Mercury paper?

Reply:
Doesn't look like it, because Mercury prescribes checking vernum of _every_ targets metadata file.

Reply:
Yes, but you get almost the same security properties because every _downloaded_ targets metadata file is checked in 4.5.2.7. When the check was done on every targets metadata file it made it impossible to recover from a fast forward attack on a delegated targets metadata file.

Reply:
Hold on, I'm confused.

When you use `root` to rotate `timestamp` and `snapshot`, you discard the previous snapshot metadata file, no? And isn't rotating `root` the first thing we do? So why can't you check vernum for every targets metadata file?

Reply:
Do you mean for the targets metadata files you **previously downloaded** or somehow checking for all targets files, even those that were not downloaded?

Reply:
All targets files. Isn't this the Mercury workflow for snapshot?

Reply:
@trishankatdatadog If a fast forward attack occurs in a delegated targets file, does root still rotate timestamp and snapshot? This may require coordination between an independent developer who owns the delegated target and the repository manager which could slow the recovery time. If the root still rotates, then the check in 4.5.2.7 could be moved back up here.

Reply:
I see. Hmm, this requires careful thinking and modeling. I want to make sure we cover all of our bases. Right now, it doesn't look like it, but maybe I'm wrong.

Reply:
Thanks for helping to clarify, @mnm678!

> I see. Hmm, this requires careful thinking and modeling. I want to make sure we cover all of our bases. Right now, it doesn't look like it, but maybe I'm wrong.

Can you elaborate, @trishankatdatadog? Why does it not look like we cover all of our bases? Maybe we should discuss this in person?

Reply:
Yes, I think we need a meeting.